api/v1alpha1/bundle_types.go

```diff
@@ -22,6 +22,8 @@ import (
 
 type BundleConditionType string
 
+type BundlePhase string
```

Comment:
I see the reasoning behind this but it looks like we end having to coerce everything to `string` in the codebase and it doesn't look that great. Is using a custom phase type beneficial for metrics somehow? 

Happy to let others weigh in.

Reply:
If we do want this, I think breaking it out into its own separate PR makes sense as well.

Reply:
So my instinct to turn this into a typed struct (BundlePhase) was to try and highlight the Phases of the Bundle resource more. The way the Reconcile is coded right now, it seems like phase transition is almost an after thought of reconciliation, whereas my argument is that phase transition is one of the primary goal of reconciliation. 
However, as I kept plumbing this I realized that just turning `Phase` from `string` to `BundlePhase` isn't enough, as highlighted by comment [here](https://github.com/operator-framework/rukpak/pull/264/files#diff-c484260008b9e8123e074e620f312d60c48375aea24e977e1dbf08c1fe13b4f6R98-R101). Reading your comment it's becoming more clear to me that it'll probably be cleaner to do that 👈🏽  first, before something like metrics emitting functionality is added, where you have to "do something before reconciliation"/"do something during reconciliation"/"do something after reconciliation" etc